sql/src/main/java/io/crate/analyze/symbol/Aggregations.java

```diff
@@ -35,6 +39,47 @@ public static boolean containsAggregation(Symbol s) {
         return AGGREGATION_SEARCHER.process(s, null);
     }
 
+    /**
+     * @return true if the symbol is an aggregation or function which contains an aggregation or a scalar.
+     */
+    public static boolean containsAggregationOrscalar(Symbol s) {
+        return AGGREGATION_OR_SCALAR_SEARCHER.process(s, null);
+    }
+
+    /**
+     * @return false if the symbol is not found in the group by, or the if one the symbol's non-literal arguments are not found in the group by.
```

Comment:
That explanation sounds a lot better - can you update the docstring to "return true if `s` has an exact match in groupBy or if the symbol is a function and at least one of the arguments is contained in the group by (recursively)"

then I guess the name is also good enough